crates/ra_parser/src/grammar/items.rs

```diff
@@ -199,6 +210,19 @@ pub(super) fn maybe_item(p: &mut Parser, m: Marker, flavor: ItemFlavor) -> Resul
         //     default fn foo() {}
         // }
 
+        // test_err default_const
+        // trait T {
+        //   default const f: u8 = 0;
+        // }
+
+        // test default_const
+        // impl T for Foo {
+        //   default const f: u8 = 0;
+        // }
+        T![const] => {
+            consts::const_def(p, m);
+        }
```

Comment:
This code is not specific to `default`, so the tests which exercise `default ` should be moved out of line, to a new file inside ` crates/ra_syntax/test_data/parser/ok/`